litellm/caching.py

```diff
@@ -1217,6 +1217,410 @@ async def async_get_cache(self, key, **kwargs):
+            collection_details = self.sync_client.get(
+                    url=f"{self.qdrant_url}/collections/{self.collection_name}",
+                    headers=self.headers
+                )
+            self.collection_info = collection_details.json()
+            print_verbose(f'Collection already exists.\nCollection details:{self.collection_info}')
+        else:
+            if quantization_config is None or quantization_config == 'binary':
+                quantization_params = {
+                    "binary": {
+                        "always_ram": False,
+                    }
+                }
+            elif quantization_config == 'scalar':
+                quantization_params = {
+                    "scalar": {
+                        "type": "int8",
+                        "quantile": 0.99,
+                        "always_ram": False
+                    }
+                }
+            elif quantization_config == 'product':
+                quantization_params = {
+                    "product": {
+                        "compression": "x16",
+                        "always_ram": False
+                    }
+                }
+            else: 
+                raise Exception("Quantization config must be one of 'scalar', 'binary' or 'product'")
+            
+            new_collection_status = self.sync_client.put(
+                url=f"{self.qdrant_url}/collections/{self.collection_name}",
+                json={
+                    "vectors": {
+                        "size": 1536,
+                        "distance": "Cosine"
+                    },
+                    "quantization_config": quantization_params
+                },
+                headers=self.headers
+            )
+            if new_collection_status.json()["result"]:
+                collection_details = self.sync_client.get(
+                        url=f"{self.qdrant_url}/collections/{self.collection_name}",
+                        headers=self.headers
+                    )
+                self.collection_info = collection_details.json()
+                print_verbose(f'New collection created.\nCollection details:{self.collection_info}')
+            else:
+                raise Exception("Error while creating new collection")
+
+    def _get_cache_logic(self, cached_response: Any):
+        if cached_response is None:
+            return cached_response
+        try:
+            cached_response = json.loads(
+                cached_response
+            )  # Convert string to dictionary
+        except:
+            cached_response = ast.literal_eval(cached_response)
+        return cached_response
+
+    def set_cache(self, key, value, **kwargs):
+        print_verbose(f"qdrant semantic-cache set_cache, kwargs: {kwargs}")
+        import uuid
+
+        # get the prompt
+        messages = kwargs["messages"]
+        prompt = ""
+        for message in messages:
+            prompt += message["content"]
```

Comment:
this will cause errors when message["content"] is list

I'd recommend using `get_formatted_prompt` from utils.py

https://github.com/BerriAI/litellm/blob/417547b6f991e05ff6f6663b783e7f8cc889ac96/litellm/utils.py#L4421

Reply:
Noted. Will look into it and update accordingly.